pkgs/common-updater/unstable-updater.nix

```diff
@@ -25,7 +27,9 @@ let
 
     # Get info about HEAD from a shallow git clone
     tmpdir="$(${coreutils}/bin/mktemp -d)"
-    ${git}/bin/git clone --bare --depth=1 "$url" "$tmpdir"
+    ${git}/bin/git clone --bare --depth=1 --single-branch \
+      ${lib.optionalString (branch != null) " --branch " + branch} \
```

Comment:
```suggestion
      --branch ${branch}
```

Reply:
FTR, as discussed on the PR, this is not a good idea because the remote repo's default branch may be something other than `master`.